src/mlpack/core/tree/rectangle_tree/rectangle_tree_impl.hpp

```diff
@@ -154,29 +155,18 @@ RectangleTree(
     dataset(deepCopy ? new MatType(*other.dataset) : &other.Dataset()),
     ownsDataset(deepCopy),
     points(other.Points()),
-    localDataset(NULL)
+    auxiliaryInfo(other.auxiliaryInfo)
 {
-  split = SplitType<RectangleTree>(other);
   if (deepCopy)
   {
     if (numChildren > 0)
     {
       for (size_t i = 0; i < numChildren; i++)
-      {
         children[i] = new RectangleTree(*(other.Children()[i]));
```

Comment:
I know this is not your code, but I wanted to see if you thought the same thing here... I think this may duplicate the dataset many times, if I do the following:

```
RTree<...> r(dataset); // makes lots of nodes
RTree<...> otherR(r, true /* deep copy */);
```

This will recursively call this constructor with `deepCopy = true`, copying the dataset every single time.  Is my understanding correct there?  If so, I'll open an issue about that separately (no need to fix it here, I think it might take a little thought).